### ISSUE TITLE
Handle float precisions

### ISSUE THREAD

**Issue by stacksmith** [cbaggers/varjo#43]
We should at least decide on how this issue should be dealt with.

**Comment by cbaggers** [comment]
True. I'm thinking we allow `the` to be used to add qualifiers to types. We already support adding qualifiers via values `(values (:smooth (v! 1 2))` so we could do the same here. The type would have to match so it would be `(the (:smooth :vec4) (v! 1 2 3 4)` 

Hmm, or maybe we treat `the` as just providing more info, so as long as it doesnt conflict with what is already known then it's ok. So `(the :smooth x)` would be ok as long as `x` hadn't already be declared to be `flat`. 

This kind of ties into a feeling that I shouldnt be handling these qualifiers in a special way at all and should instead rebuild them using the newer metadata system. Then `(let ((x (v! 1 2))) (declare (flat) x) x)` would be valid and `the` support can just fall out of that
